lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -0,0 +1,262 @@
+        context.read<SummarySettingsModel>().setMaxTokens(maxTokens);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _maxTokensController,
+            onChanged: (value) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_maxTokensController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 500',
+            ),
+          ),
+          const SizedBox(height: 16),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Unlimited',
+              ),
+              Switch(
+                value: _maxTokensController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class RandomSeedDialog extends StatefulWidget {
+  const RandomSeedDialog({
+    super.key,
+  });
+
+  @override
+  State<RandomSeedDialog> createState() => _RandomSeedDialogState();
+}
+
+class _RandomSeedDialogState extends State<RandomSeedDialog> {
+  final TextEditingController _randomSeedController = TextEditingController();
+
+  @override
+  void initState() {
+    _randomSeedController.text =
+        context.read<SummarySettingsModel>().settings.randomSeed?.toString() ??
+            '';
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _randomSeedController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CommonSettingsDialog(
+      title: 'Random Seed',
+      description: 'Write a number if you want to set a random seed',
+      onSettingChanged: () {
+        final randomSeed = _randomSeedController.text.isEmpty
+            ? null
+            : int.tryParse(_randomSeedController.text);
+        context.read<SummarySettingsModel>().setRandomSeed(randomSeed);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _randomSeedController,
+            onChanged: (_) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_randomSeedController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 123',
+            ),
+          ),
```

Comment:
I would add `autofocus` on this one. It will make the keyboard appear automatically on mobile.